seabird_chat_ingest.proto

```diff
@@ -0,0 +1,164 @@
+syntax = "proto3";
+
+option go_package = ".;pb";
+
+import "common.proto";
+
+package seabird;
+
+message ChatUser {
+  string id = 1;
+  string name = 2;
+}
+
+message ChatChannel {
+  string id = 1;
+  string name = 2;
+}
+
+// When a chat service receives a PingChatRequest, it MUST respond with a
+// PongChatEvent containing the copied nonce.
+message PingChatRequest {
+  string nonce = 1;
+}
+
+// PongChatRequest will be sent in response to a PingChatEvent.
+message PongChatRequest {
+  string nonce = 1;
+}
+
+message SendMessageChatRequest {
+  string channel_id = 1;
+  string message = 2;
+}
+
+message SendPrivateMessageChatRequest {
+  string user_id = 1;
+  string message = 2;
+}
+
+message JoinChannelChatRequest {
+  string channel_name = 1;
+}
+
+message LeaveChannelChatRequest {
+  string channel_id = 1;
+}
+
+message SetChannelTopicChatRequest {
+  string channel = 1;
+  string topic = 2;
+}
+
+// Each ChatRequest has an ID (which can be attached to outgoing events - only
+// the first outgoing event with this ID will be used as the reply) and an inner
+// message type. Having an ID allows us to route an event back to the requestor
+// without having the chat backend implement a service.
+message ChatRequest {
+  string id = 1;
+
+  oneof inner {
+    PingChatRequest ping = 2;
+    PongChatRequest pong = 3;
+    SendMessageChatRequest send_message = 4;
+    SendPrivateMessageChatRequest send_private_message = 5;
+    JoinChannelChatRequest join_channel = 6;
+    LeaveChannelChatRequest leave_channel = 7;
+    SetChannelTopicChatRequest set_channel_topic = 8;
+  }
+}
+
+// HelloChatEvent must be the first event sent by the chat backend. If a different
+// event is sent, the connection will be closed.
+message HelloChatEvent {
+  common.Identity identity = 1;
+}
```

Comment:
Is this the only part of the handshake? Does it have both a request and a response associated with it? What about reconnects? I don't see another event type that answers these questions, but I may be overlooking something.

Reply:
Currently, yes. This message is here so the identity token can be sent and so when we want to add more information to it (like all the channels we're currently in, stuff like that) there's a good place to put it.